nomad/structs/structs.go

```diff
@@ -8136,7 +8144,15 @@ func (t *Task) Warnings() error {
 	for idx, tmpl := range t.Templates {
 		if err := tmpl.Warnings(); err != nil {
 			err = multierror.Prefix(err, fmt.Sprintf("Template[%d]", idx))
-			mErr.Errors = append(mErr.Errors, err)
+			mErr = *multierror.Append(&mErr, err)
+		}
+	}
+
+	// Validate task-level services.
+	for _, s := range t.Services {
+		if err := s.Warnings(); err != nil {
+			err = multierror.Prefix(err, fmt.Sprintf("Service %q:", s.Name))
+			mErr = *multierror.Append(&mErr, err)
```

Comment:
This way of appending errors flattens inner errors so the output goes from something like this:

```
Group "web" has warnings: 2 errors occurred:
        * 1 error occurred:
        * Service "web": identities without an expiration are insecure


        * 2 errors occurred:
        * Task "web": IOPS has been deprecated as of Nomad 0.9.0. Please remove IOPS from resource block.
        * Task "web": Service "web": identities without an expiration are insecure
```

To this:

```
Group "web" has warnings: 2 errors occurred:
        * Service "web": identities without an expiration are insecure
        * Task "web" has warnings: 2 errors occurred:
        * IOPS has been deprecated as of Nomad 0.9.0. Please remove IOPS from resource block.
        * Service "web": identities without an expiration are insecure
```

Still weird, but I think a little better?